src/programs/azimuthal_average/azimuthal_average.cpp

```diff
@@ -0,0 +1,1534 @@
+
+	// Profiling
+	wxDateTime	overall_start = wxDateTime::Now();
+	wxDateTime 	overall_finish;
+	wxDateTime 	read_frames_start;
+	wxDateTime	read_frames_finish;
+	wxDateTime	first_alignment_start;
+	wxDateTime	first_alignment_finish;
+	wxDateTime	main_alignment_start;
+	wxDateTime	main_alignment_finish;
+
+	long pixel_counter; // not used
+	long image_counter;
+
+	// rotational and translational alignment of images
+	Image *image_stack = new Image[number_of_input_images];
+	// Arrays to hold the shifts and rotations
+	float *x_shifts = new float[number_of_input_images];
+	float *y_shifts = new float[number_of_input_images];
+	float *psi_angles = new float[number_of_input_images];
+	float min_shift_in_pixels;
+	float max_shift_in_pixels;
+	float termination_threshold_in_pixels;
+	float unitless_bfactor;
+
+	// read CTF from file
+	if (input_ctf_values_from_text_file == true)
+	{
+		for (image_counter = 0; image_counter < number_of_input_images; image_counter++)
+		{
+
+			input_text->ReadLine(temp_float);
+
+			if (input_text->records_per_line == 3)
+			{
+				ctf_parameters[image_counter][0] = acceleration_voltage;
+				ctf_parameters[image_counter][1] = spherical_aberration;
+				ctf_parameters[image_counter][2] = amplitude_contrast;
+				ctf_parameters[image_counter][3] = temp_float[0];
+				ctf_parameters[image_counter][4] = temp_float[1];
+				ctf_parameters[image_counter][5] = temp_float[2];
+				ctf_parameters[image_counter][6] = 0.0;
+				ctf_parameters[image_counter][7] = 0.5;
+				ctf_parameters[image_counter][8] = 0.0;
+				ctf_parameters[image_counter][9] = pixel_size;
+				ctf_parameters[image_counter][10] = 0.0;
+			}
+			else
+			{
+				ctf_parameters[image_counter][0] = acceleration_voltage;
+				ctf_parameters[image_counter][1] = spherical_aberration;
+				ctf_parameters[image_counter][2] = amplitude_contrast;
+				ctf_parameters[image_counter][3] = temp_float[0];
+				ctf_parameters[image_counter][4] = temp_float[1];
+				ctf_parameters[image_counter][5] = temp_float[2];
+				ctf_parameters[image_counter][6] = 0.0;
+				ctf_parameters[image_counter][7] = 0.5;
+				ctf_parameters[image_counter][8] = 0.0;
+				ctf_parameters[image_counter][9] = pixel_size;
+				ctf_parameters[image_counter][10] = temp_float[3];
+			}
+		}
+	}
+
+	// read in image stack and FFT
+
+	read_frames_start = wxDateTime::Now();
+	// input stack times ctf
+	Image *image_stack_times_ctf = new Image[number_of_input_images];
+	// allocate arrays for the ctf sum of squares
+	float *ctf_sum_of_squares;
+
+	// read image stack
+
+	for (image_counter = 0; image_counter < number_of_input_images; image_counter++)
+	{
+		// Read from disk
+		image_stack[image_counter].ReadSlice(&my_input_file, image_counter + 1);
+		image_stack_times_ctf[image_counter].ReadSlice(&my_input_file, image_counter + 1);
+
+		// Normalize images
+		normalize_image(&image_stack[image_counter], pixel_size, 10.0);
+		normalize_image(&image_stack_times_ctf[image_counter], pixel_size, 10.0);
```

Comment:
image_stack[image_counter] and image_stack_times_ctf[image_counter] are identical at this point, so why call normalize_image twice?

Reply:
This is true and it looks like my approach is not optimal. Instead, I will first read the images and store in image_stack, normalize, FFT, and populate image_stack_times_ctf by copying it from image_stack since they are identical at this point.